Lib/posixpath.py

```diff
@@ -263,7 +263,7 @@ def expanduser(path):
             return path
         name = path[1:i]
         if isinstance(name, bytes):
```

Comment:
This `isinstance()` check is unnecessary; `os.fsdecode` returns a `str` argument unchanged (this is a documented feature: https://docs.python.org/3/library/os.html#os.fsdecode). We can simply do `name = os.fsdecode(path[1:i])`

Reply:
Good point. On other hand, it makes the code slower for strings (which are more common than bytes):
```
$ ./python -m timeit -s "import os" "x = 'abc'" "if isinstance(x, bytes): x = os.fsdecode(x)"
5000000 loops, best of 5: 49.4 nsec per loop
$ ./python -m timeit -s "import os" "x = 'abc'" "x = os.fsdecode(x)"
2000000 loops, best of 5: 101 nsec per loop
$ ./python -m timeit -s "import os" "x = b'abc'" "if isinstance(x, bytes): x = os.fsdecode(x)"
2000000 loops, best of 5: 167 nsec per loop
$ ./python -m timeit -s "import os" "x = b'abc'" "x = os.fsdecode(x)"
2000000 loops, best of 5: 145 nsec per loop
```
And the `ntpath` implementation also uses `isinstance()` in similar place.

So I will keep this micro-optimization for now.